File: platforms/kotlin/examples/scene/material/occlusion_texture.kt
import org.fragmentcolor.*

val renderer = Renderer()
val ao_bytes = listOf(220.0f, 0.0f, 0.0f, 255.0f, 180.0f, 0.0f, 0.0f, 255.0f, 200.0f, 0.0f, 0.0f, 255.0f, 160.0f, 0.0f, 0.0f, 255.0f, .0f)
val ao = renderer.createTexture(TextureInputMobile.Bytes(ao_bytes.let { ba -> ByteArray(ba.size) { i -> ba[i].toInt().and(0xFF).toByte() } }), null)
val mat = Material.pbr().occlusionTexture(ao)

File: platforms/swift/examples/scene/material/occlusion_texture.swift
import FragmentColor

let renderer = Renderer()
let ao_bytes = [
    220, 0, 0, 255,
    180,   0, 0, 255,
    200,   0, 0, 255,
    160,   0, 0, 255,
]
let ao = try await renderer.createTexture((ao_bytes, [2, 2]))
let mat = Material.pbr().occlusionTexture(ao)